t/Util.pm

```diff
@@ -220,10 +220,12 @@ sub spawn_h2o {
     # setup the configuration file
     $conf = $conf->($port, $tls_port)
         if ref $conf eq 'CODE';
+    my $user = $< == 0 ? "user: root" : "";
```

Comment:
@kazuho 

BTW I think this default behavior in Util.pm may be harmful because it unexpectedly retains the root privilege in tests. So every test that wants `user: root` must set it in their own config. What do you think?

Reply:
This is intentional for various reasons:
* When the user runs `make check` as root, then that user has committed to running arbitrary code as root.
* In practice, it is hard to determine a non-root user that we can use for running tests. `nobody` might not exist (especially in containers). Current directory might not be accessible by `nobody`.

Reply:
I see. Thanks for the explanation. Either makes sense.